<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Modeling spatially varying gene correlation across 2D space • spCorr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Modeling spatially varying gene correlation across 2D space">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">spCorr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/spCorr-2D.html">Modeling spatially varying gene correlation across 2D space</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Modeling spatially varying gene correlation
across 2D space</h1>
                        <h4 data-toc-skip class="author"></h4>
<p><strong>Chenxin
(Flora) Jiang</strong><br>
Department of Statistics and Data Science, University of California, Los
Angeles<br><a href="mailto:cflorajiang@g.ucla.edu" class="email">cflorajiang@g.ucla.edu</a></p>
            
            <h4 data-toc-skip class="date">21 October 2025</h4>
      

      <div class="hidden name"><code>spCorr-2D.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">spCorr</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This tutorial demonstrates how to use <strong>spCorr</strong> to (i)
infer spot-level gene–gene correlations for specified gene pairs, and
(ii) identify gene pairs exhibiting <em>spatially varying correlation
(SVC)</em> patterns across a 2D spatial space.</p>
</div>
<div class="section level2">
<h2 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h2>
<p>In this example, we analyze a subset of the <strong>human oral
squamous cell carcinoma (OSCC)</strong> dataset generated using the
<strong>10x Visium</strong> platform (sample 2 from <a href="https://www.nature.com/articles/s41467-023-40271-4" class="external-link">Arora <em>et
al.</em>, 2023</a>). We focus on transcription factor (TF)–target gene
pairs curated from the <a href="https://www.grnpedia.org/trrust/" class="external-link">TRRUST
v2 database</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load example spatial transcriptomics data</span></span>
<span><span class="co"># It contains two objects: counts (gene expression matrix) and cov_mat (spot-level covariates)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/58905268"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># counts: gene expression count matrix (genes x spots)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 15624  1747</span></span>
<span></span>
<span><span class="co"># cov_mat: spot spatial coordinates and other spot-level covariates</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cov_mat</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1747    5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cov_mat</span><span class="op">)</span></span>
<span><span class="co">#&gt;                       x1    x2 cluster_annotations         pathologist_anno.x</span></span>
<span><span class="co">#&gt; AAACATTTCCCGGATT-2 17022 16020                core                        SCC</span></span>
<span><span class="co">#&gt; AAACCGGGTAGGTACC-2 12218  6043                  nc Lymphocyte Negative Stroma</span></span>
<span><span class="co">#&gt; AAACCGTTCGTCCAGG-2 14740  8064                edge                        SCC</span></span>
<span><span class="co">#&gt; AAACCTAAGCAGCCGG-2 18025 13992                  nc                        SCC</span></span>
<span><span class="co">#&gt; AAACGAGACGGTTGAT-2 10469 13427                edge                        SCC</span></span>
<span><span class="co">#&gt; AAACGGGCGTACGGGT-2 18027 15149          transitory                        SCC</span></span>
<span><span class="co">#&gt;                    tumor_annotations</span></span>
<span><span class="co">#&gt; AAACATTTCCCGGATT-2             tumor</span></span>
<span><span class="co">#&gt; AAACCGGGTAGGTACC-2                nc</span></span>
<span><span class="co">#&gt; AAACCGTTCGTCCAGG-2             tumor</span></span>
<span><span class="co">#&gt; AAACCTAAGCAGCCGG-2             tumor</span></span>
<span><span class="co">#&gt; AAACGAGACGGTTGAT-2             tumor</span></span>
<span><span class="co">#&gt; AAACGGGCGTACGGGT-2             tumor</span></span>
<span></span>
<span><span class="co">## Load TF–target gene pairs</span></span>
<span><span class="va">tf_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/58905265"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tf_df</span> <span class="op">&lt;-</span> <span class="va">tf_df</span><span class="op">[</span><span class="va">tf_df</span><span class="op">$</span><span class="va">type</span><span class="op">==</span><span class="st">"Activation"</span> <span class="op">|</span> <span class="va">tf_df</span><span class="op">$</span><span class="va">type</span><span class="op">==</span><span class="st">"Unknown"</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tf_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;      tf target       type               id</span></span>
<span><span class="co">#&gt; 2  AATF CDKN1A    Unknown         17157788</span></span>
<span><span class="co">#&gt; 4  AATF    MYC Activation         20549547</span></span>
<span><span class="co">#&gt; 6  ABL1    BAX Activation         11753601</span></span>
<span><span class="co">#&gt; 9  ABL1  CCND2 Activation         15509806</span></span>
<span><span class="co">#&gt; 10 ABL1 CDKN1A Activation 11753601;9916993</span></span>
<span><span class="co">#&gt; 14 ABL1    JUN Activation         15145216</span></span></code></pre></div>
<p>Now we prepare the input data for <strong>spCorr</strong>
analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create gene pair list (data frame format)</span></span>
<span><span class="va">gene_pair_list</span> <span class="op">&lt;-</span> <span class="va">tf_df</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tf"</span>, <span class="st">"target"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gene_pair_list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">gene_pair_list</span>, <span class="fl">1</span>, <span class="va">paste</span>, collapse <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the unique set of genes from these pairs</span></span>
<span><span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gene_pair_list</span><span class="op">$</span><span class="va">tf</span>, <span class="va">gene_pair_list</span><span class="op">$</span><span class="va">target</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of selected gene pairs:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gene_pair_list</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of selected gene pairs: 1316</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of unique genes:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_list</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of unique genes: 601</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_pair_list</span><span class="op">)</span></span>
<span><span class="co">#&gt;               tf target</span></span>
<span><span class="co">#&gt; AATF_CDKN1A AATF CDKN1A</span></span>
<span><span class="co">#&gt; AATF_MYC    AATF    MYC</span></span>
<span><span class="co">#&gt; ABL1_BAX    ABL1    BAX</span></span>
<span><span class="co">#&gt; ABL1_CCND2  ABL1  CCND2</span></span>
<span><span class="co">#&gt; ABL1_CDKN1A ABL1 CDKN1A</span></span>
<span><span class="co">#&gt; ABL1_JUN    ABL1    JUN</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-spcorr">Run spCorr<a class="anchor" aria-label="anchor" href="#run-spcorr"></a>
</h2>
<p>We now apply <strong>spCorr</strong> to infer spot-level gene–gene
correlations for the selected gene pairs and to identify gene pairs with
spatially varying correlation (SVC) pattern across the 2D spatial
space.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spCorr.html">spCorr</a></span><span class="op">(</span><span class="va">counts</span>,</span>
<span>             <span class="va">gene_list</span>,</span>
<span>             <span class="va">gene_pair_list</span>,</span>
<span>             <span class="va">cov_mat</span>,</span>
<span>             formula1 <span class="op">=</span> <span class="st">"tumor_annotations"</span>,</span>
<span>             family1 <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>             formula2 <span class="op">=</span> <span class="st">"s(x1, x2, bs='tp', k=30)"</span>,</span>
<span>             family2 <span class="op">=</span> <span class="fu"><a href="../reference/quasiproductr.html">quasiproductr</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>             DT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>             global_test <span class="op">=</span> <span class="st">"lrt"</span>,</span>
<span>             return_models <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             return_coefs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             check_morani <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             preconstruct_smoother <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             ncores <span class="op">=</span> <span class="fl">10</span>,</span>
<span>             seed <span class="op">=</span> <span class="fl">123</span><span class="op">)</span></span>
<span><span class="co">#&gt; Start Marginal Fitting for 601 genes</span></span>
<span><span class="co">#&gt; Start Extracting Spatially Varying Gene Pairs</span></span>
<span><span class="co">#&gt; Start Product Fitting for 1316 gene pairs</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">res</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 6</span></span>
<span><span class="co">#&gt;  $ res_global  : Named num [1:1316] 0.483 0.0795 0.2519 0.42 0.5005 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:1316] "AATF_CDKN1A" "AATF_MYC" "ABL1_BAX" "ABL1_CCND2" ...</span></span>
<span><span class="co">#&gt;  $ edf         : Named num [1:1316] 13.1 17.3 20.3 12.2 14.7 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:1316] "AATF_CDKN1A" "AATF_MYC" "ABL1_BAX" "ABL1_CCND2" ...</span></span>
<span><span class="co">#&gt;  $ res_local   : num [1:1316, 1:1747] 0.226 0.67 0.57 0.633 0.233 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;  $ res_local_pi:List of 1316</span></span>
<span><span class="co">#&gt;  $ residuals   : num [1:601, 1:1747] 0.237 0.251 0.231 0.244 0.201 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;  $ marginals   : num [1:601, 1:1747] -0.716 -0.67 -0.737 -0.694 -0.837 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-spot-level-correlations">Visualize spot-level correlations<a class="anchor" aria-label="anchor" href="#visualize-spot-level-correlations"></a>
</h2>
<p>We next visualize the inferred spot-level correlations and SVC
patterns for specific gene pairs. The spot-level correlation estimates
are stored in <code>res$res_local</code>, which is a spot-by-gene pair
matrix.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specify gene pairs to visualize</span></span>
<span><span class="va">gene_pair</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ELF3_SPRR1B"</span>, <span class="st">"ELF3_SPRR3"</span>, <span class="st">"ELF3_SPRR2A"</span>, <span class="st">"ELF3_SPRR1A"</span>, <span class="st">"ELF3_CLDN7"</span><span class="op">)</span></span>
<span><span class="va">rho_mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">res_local</span><span class="op">[</span><span class="va">gene_pair</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span><span class="op">)</span>  <span class="co"># spots × gene pairs</span></span>
<span></span>
<span><span class="co"># Combine with covariates and reshape to long format</span></span>
<span><span class="va">plot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">cov_mat</span>, <span class="va">rho_mat</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">all_of</a></span><span class="op">(</span><span class="va">gene_pair</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"gene_pair"</span>, values_to <span class="op">=</span> <span class="st">"rho"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_corr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plot_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x2</span>, y <span class="op">=</span> <span class="op">-</span><span class="va">x1</span>, color <span class="op">=</span> <span class="va">rho</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridis/reference/viridis_pal.html" class="external-link">viridis_pal</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"magma"</span><span class="op">)</span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>, name <span class="op">=</span> <span class="st">"correlation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span>ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">gene_pair</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="cn">NA</span>, linewidth <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,</span>
<span>    strip.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="cn">NA</span>, linewidth <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">p_corr</span></span></code></pre></div>
<p><img src="spCorr-2D_files/figure-html/p_corr-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="identify-gene-pairs-with-significant-svc-patterns">Identify gene pairs with significant SVC patterns<a class="anchor" aria-label="anchor" href="#identify-gene-pairs-with-significant-svc-patterns"></a>
</h2>
<p>The <strong>spCorr</strong> also provides SVC testing to identify
gene pairs whose correlations vary significantly across spatial
locations. The BH-adjusted <em>p</em>-values from the SVC testing are
stored in <code>res$res_global</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pval_svc</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">res_global</span></span>
<span><span class="va">gene_pair_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">pval_svc</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of gene pairs with significant SVC patterns (FDR &lt; 0.05):"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gene_pair_sig</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of gene pairs with significant SVC patterns (FDR &lt; 0.05): 294</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] viridis_0.6.5     viridisLite_0.4.2 tidyr_1.3.1       dplyr_1.1.4      </span></span>
<span><span class="co">#&gt; [5] ggplot2_4.0.0     spCorr_0.0.0.9000 BiocStyle_2.26.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1    xfun_0.53           bslib_0.9.0        </span></span>
<span><span class="co">#&gt;  [4] purrr_1.1.0         splines_4.2.3       lattice_0.22-6     </span></span>
<span><span class="co">#&gt;  [7] vctrs_0.6.5         generics_0.1.4      htmltools_0.5.8.1  </span></span>
<span><span class="co">#&gt; [10] yaml_2.3.10         mgcv_1.9-3          rlang_1.1.6        </span></span>
<span><span class="co">#&gt; [13] pkgdown_2.1.3       jquerylib_0.1.4     pillar_1.11.1      </span></span>
<span><span class="co">#&gt; [16] glue_1.8.0          withr_3.0.2         RColorBrewer_1.1-3 </span></span>
<span><span class="co">#&gt; [19] S7_0.2.0            lifecycle_1.0.4     gtable_0.3.6       </span></span>
<span><span class="co">#&gt; [22] ragg_1.5.0          htmlwidgets_1.6.4   evaluate_1.0.5     </span></span>
<span><span class="co">#&gt; [25] labeling_0.4.3      knitr_1.50          fastmap_1.2.0      </span></span>
<span><span class="co">#&gt; [28] parallel_4.2.3      Rcpp_1.1.0          scales_1.4.0       </span></span>
<span><span class="co">#&gt; [31] BiocManager_1.30.26 cachem_1.1.0        desc_1.4.3         </span></span>
<span><span class="co">#&gt; [34] jsonlite_2.0.0      farver_2.1.2        systemfonts_1.3.1  </span></span>
<span><span class="co">#&gt; [37] fs_1.6.6            gridExtra_2.3       textshaping_1.0.4  </span></span>
<span><span class="co">#&gt; [40] digest_0.6.37       bookdown_0.45       grid_4.2.3         </span></span>
<span><span class="co">#&gt; [43] cli_3.6.5           tools_4.2.3         magrittr_2.0.4     </span></span>
<span><span class="co">#&gt; [46] sass_0.4.10         tibble_3.3.0        dichromat_2.0-0.1  </span></span>
<span><span class="co">#&gt; [49] ape_5.8-1           pkgconfig_2.0.3     Matrix_1.6-5       </span></span>
<span><span class="co">#&gt; [52] rmarkdown_2.30      rstudioapi_0.17.1   R6_2.6.1           </span></span>
<span><span class="co">#&gt; [55] nlme_3.1-164        compiler_4.2.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chenxin (Flora) Jiang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
